tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -129,12 +150,44 @@ def tags_route(self, request):
     @wrappers.Request.application
     def scalars_route(self, request):
         """Given a tag and single run, return array of ScalarEvents."""
-        tag = request.args.get("tag")
-        run = request.args.get("run")
+        if request.method == "GET":
+            tag = request.args.get("tag")
+            run = request.args.get("run")
+        else:
+            response = (
+                "%s requests are forbidden by the scalars plugin."
+                % request.method
+            )
+            return http_util.Respond(request, response, "text/plain", code=405)
```

Comment:
HTTP 405 responses are required to contain an `Allow` header listing the
allowed methods. FWIW, given that you’ve moved the POST handler to a
separate route, it’s fine with me if you want to omit the 405 checking
altogether (few other routes implement it). But if you do implement it,
please conform to spec.

(Also, if you do implement it, it seems confusing to say that GET/POST
requests are “forbidden by the scalars plugin” in both cases. Clearly
neither GET nor POST requests are forbidden by the plugin in general.)